### PR TITLE
#157470184 Users should be able to log out 

### PR DESCRIPTION
**Pull request by Billmike**
#### What does this PR do?
Create the log-out functionality for activo application.

#### Description of tasks to be completed?
- Create profile drop-down component
- Create an interface for the drop-down component
- Write styles for the component
- Write logic for logging out user

#### How should this be manually tested?
- After cloning the application, run yarn install to install the required dependencies for the application.
- The application should open on activo-dev.andela.com:3000.
- Log in with your Andela account. You should be re-directed to the dashboard page.
- Next to your displayed name, click on the caret. A drop-down should be displayed with a text of `LOG OUT`.
- Click on it. You should now be re-directed to the authentication page.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#157470184](https://www.pivotaltracker.com/story/show/157470184)

#### Screenshots
screen shot 2018-06-12 at 3 17 42 pm

screen shot 2018-06-12 at 3 26 33 pm

### CODE REVIEW

File: src/components/NavBar/NavBar.scss [hunk 1]
```diff
@@ -117,7 +117,7 @@
     padding-right: 9px;
   }
 
-  .settings-icon {	
+  .settings-icon {
     height: 22px;
```
Comment (andela-kakpobome) [inline src/components/NavBar/NavBar.scss]: Which should be using strictly `rem` not `px`

File: src/components/NavBar/index.tsx [hunk 2]
```diff
@@ -28,6 +33,15 @@ export class NavBar extends React.Component<NavBarProps, {}> {
...
 
+  /**
+   * This handler changes the state of the dropDownHidden key
+   * in the state object.
+   * 
+   * @returns {object} An updated version of the un-mutated state
+   */
+  toggleDropDownState = () => {
+    this.setState({ dropDownHidden: !this.state.dropDownHidden });
+  }
```
Comment (andela-kakpobome) [inline src/components/NavBar/index.tsx]: Leave a space after this line

File: src/components/NavBar/index.tsx [hunk 3]
```diff
@@ -51,17 +65,18 @@ export class NavBar extends React.Component<NavBarProps, {}> {
...
                   />
-                  <i className="drop-down-icon">&#9662;</i>
+                  <i onClick={this.toggleDropDownState} className="drop-down-icon">&#9662;</i>
+                  <ProfileDropDown hidden={this.state.dropDownHidden} />
                 </div>
               </div>
             </div>
 
-            <span className="line"></span>
+            <span className="line" />
```
Comment (andela-kakpobome) [inline src/components/NavBar/index.tsx]: This is not a component, its an actual HTML element and deserves it's own closing tag

File: src/components/ProfileDropDown/interfaces.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface IProfileDropDownProps {
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/interfaces.d.ts]: Remove the `I` prefix

File: src/components/ProfileDropDown/ProfileDropDown.scss [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    border: solid 0.05rem #E3E3E3;
+    font-size: 0.875rem;
+    font-weight: 300;
+    text-align: center;
+    text-transform: uppercase;
+    border-radius: 0.2rem;
+    box-shadow: 0 1rem 2rem 0 rgba(0, 0, 0, 0.1);
+    transition:height 2000ms ease-in;
+  }
+  .drop-down-link {
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.scss]: Line space before each rule.

File: src/components/ProfileDropDown/ProfileDropDown.scss [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+.drop-down-menu {
+  position: absolute;
+  margin-left: rem(40px);
+  margin-top: rem(30px);
+
+  .drop-down-item {
+    position: absolute;
+    z-index: 10;
+    background-color: #FDFDFD;
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.scss]: Why are these colors not being referred from the palette? That's the whole point of having a palette.

File: src/components/ProfileDropDown/ProfileDropDown.scss [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+.drop-down-icon {
+  cursor: pointer;
+}
+
+.drop-down-menu {
+  position: absolute;
+  margin-left: rem(40px);
+  margin-top: rem(30px);
+
+  .drop-down-item {
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.scss]: Let's continue with the BEM methodology. Given that the `.drop-down-menu` is the block, this is one of the elements, `item`. So this should be `__item`, so the resulting class becomes `.drop-down-icon__item` same with `link`.

Dropdown should be one word. i.e `dropdown`

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+// react libraries
+import * as React from 'react';
+
+// third-party libraries
+import { shallow } from 'enzyme';
+import toJSON from 'enzyme-to-json';
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.test.tsx]: What's the point of this import? I don't see it being used anywhere. Please clarify.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+// component
+import ProfileDropDown from './index';
+
+describe('<ProfileDropDown />', () => {
+  const props = {
+    hidden: false,
+  };
+  it('should render the drop-down component correctly', () => {
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Let there be a line space before this & dropdown is one word

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+// react libraries
+import * as React from 'react';
+
+// third-party libraries
+import { shallow } from 'enzyme';
+import toJSON from 'enzyme-to-json';
+
+// component
+import ProfileDropDown from './index';
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.test.tsx]: This should be `./` not `./index`

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('<ProfileDropDown />', () => {
+  const props = {
+    hidden: false,
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.test.tsx]: How about when it's true? Don't see a test for that

File: src/components/ProfileDropDown/index.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import { Link } from 'react-router-dom';
+
+// styles
+import './ProfileDropDown.scss';
+
+// interfaces
+import { IProfileDropDownProps } from './interfaces';
+
+// helper functions
+import { authService } from '../../utils/auth';
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/index.tsx]: Use the alias

File: src/components/AuthenticatedRoute/AuthenticatedRoute.test.tsx [hunk 13]
```diff
@@ -10,7 +10,7 @@ import { mount, shallow } from 'enzyme';
 import AuthenticatedRoute from './index';
 
 describe('The AuthenticatedRoute component', () => {
-  const TestComponent = () => <div></div>;
+  const TestComponent = () => <div />;
```
Comment (andela-kakpobome) [inline src/components/AuthenticatedRoute/AuthenticatedRoute.test.tsx]: `Div` is not a component, it's a first class HTML element, and deserves it's own closing block

File: src/components/NavBar/NavBar.scss (hunk 1 above)
Comment (Billmike) [reply src/components/NavBar/NavBar.scss]: Fixed.

File: src/components/NavBar/index.tsx (hunk 3 above)
Comment (Billmike) [reply src/components/NavBar/index.tsx]: Fixed.

File: src/components/ProfileDropDown/interfaces.d.ts (hunk 4 above)
Comment (Billmike) [reply src/components/ProfileDropDown/interfaces.d.ts]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.scss (hunk 5 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.scss]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.scss (hunk 6 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.scss]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx (hunk 8 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.test.tsx]: I was initially using it to convert the snapshot to a readable JSON, but I forgot to remove it. It's been fixed now.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx (hunk 9 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx (hunk 10 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx (hunk 11 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Fixed.

File: src/components/ProfileDropDown/index.tsx (hunk 12 above)
Comment (Billmike) [reply src/components/ProfileDropDown/index.tsx]: Fixed.

File: src/components/AuthenticatedRoute/AuthenticatedRoute.test.tsx (hunk 13 above)
Comment (Billmike) [reply src/components/AuthenticatedRoute/AuthenticatedRoute.test.tsx]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.scss (hunk 7 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.scss]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.scss [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+  &__link {
+    text-decoration: none;
+    color: map-get($grey, 600)
+  }
+
+  &__link:hover {
+    color: map-get($palette, primary);
+  }
+
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.scss]: Lose this

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+// third-party libraries
+import { shallow } from 'enzyme';
+
+// component
+import ProfileDropDown from './';
+
+const props = {
+  hidden: false,
+};
```
Comment (andela-kakpobome) [inline src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Let this be defined within the describe

File: src/components/ProfileDropDown/ProfileDropDown.scss (hunk 14 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.scss]: Fixed.

File: src/components/ProfileDropDown/ProfileDropDown.test.tsx (hunk 15 above)
Comment (Billmike) [reply src/components/ProfileDropDown/ProfileDropDown.test.tsx]: Fixed.

**Comment by Billmike** [comment]
@bourgeois247 Merge conflicts fixed.

**Pull request merged**